test/local/schemas/functional.test.ts

```diff
@@ -0,0 +1,91 @@
+import { FunctionalSchema } from "@brink-sdk/intents/schemas/functional";
+import { DeclarationArgs } from "@brinkninja/types";
+import { expect } from 'chai';
+
+
+describe('FunctionalSchema', () => {
+  it('should validate a valid declaration', () => {
+    const { error } = FunctionalSchema.validate(declarationArgs);
+    expect(error).to.be.undefined;
+  });
+
+  it('should throw meaningful error for invalid segment params', () => {
+    let value = declarationArgs
+    // @ts-ignore
+    value.intents[0].segments[0].params = { index: '0' }; // missing value
+    const { error } = FunctionalSchema.validate(value);
+    expect(error.message).to.match(/params.value.*is required/);
```

Comment:
does this error tell me which missing intent/segment/param is required?

Reply:
True. I made the assertion more specific.